osvr/RenderKit/RenderManagerBase.cpp

```diff
@@ -2870,8 +2872,23 @@ namespace renderkit {
                     p2.m_distortionParameters[eye].m_distortionCOP[1] = scaled;
                 }
 
+                // If we've been asked for asynchronous time warp, we layer
+                // the request on top of a request for a DirectRender instance
+                // to harness.  @todo This should be doable on top of a non-
+                // DirectMode interface as well.
                 std::unique_ptr<RenderManagerD3D11Base> host = nullptr;
-                host.reset(openRenderManagerDirectMode(context, p2));
+                if (p.m_asynchronousTimeWarp) {
+                  RenderManager::ConstructorParameters pTemp = p2;
+                  pTemp.m_graphicsLibrary.D3D11 = nullptr;
+                  auto wrappedRm = openRenderManagerDirectMode(context, pTemp);
+                  auto atwRm = new RenderManagerD3D11ATW(context, p2, wrappedRm);
+                  host.reset(atwRm);

```

Comment:
This line and the previous one should be combined into one, for exception safety.